roles/dns-config/tasks/main.yml

```diff
@@ -1,4 +1,11 @@
 ---
+- name: assert that variables are defined
+  fail:
+    msg: "Variable '{{ item }}' is not defined"
+  when: item not in vars
+  with_items:
+    - dns_config_servers
```

Comment:
Can we leave the target file unmodified in this case instead of failing? It requires someone to edit configuration vs providing sane defaults.

Reply:
I'd argue there are some scenarios where there are no sane defaults. `dns_config_servers` and `kerberos_client_realm_name` seem like two of those scenarios. `ls` has a sane default arg (`.`), but what's a sane default for `ping`?

This playbook would never be included in a generic `*-cluster.yml` playbook, anyway. I think it's OK to have this one fail loudly.